app/Services/Manager.php

```diff
@@ -322,36 +323,92 @@ public function appendReportback($data, $parameters)
 
     /**
      * Get the top three reportbacks from a given leaderboard.
-     * Adds custom info about each placement for display.
+     * Adds custom info based on options passed in.
      *
-     * @param array $leaderboard
+     * @param array $leaderboard, array $options
      * @return array $topThree
      */
-    public function getTopThreeReportbacks($leaderboard)
+    public function getTopThreeReportbacks($leaderboard, $options = [])
     {
         $topThreeUsers = array_slice($leaderboard, 0, 3);
         $places = ['1st', '2nd', '3rd'];
         $prizeCopy = ['Taking home the $100 AMEX gift card', 'Winner of the $50 amex gift card', 'Winner of the $25 amex gift card'];
 
         $topThree = [];
+        $includeUserIds = false;
+        $hasCompetitionId = false;
+        $hasMessageId = false;
 
-        foreach ($topThreeUsers as $key => $user) {
-            $reportbackItems = $user->reportback->reportback_items->data;
-            $latestReportbackItem = array_pop($reportbackItems);
+        if (isset($options['includeUserIds']) && $options['includeUserIds']) {
+            $includeUserIds = true;
```

Comment:
I don't think you need to set a different variable here if it was already passed into the `$options` array instead, you could just replace the if statement on line 361 to with the if clause you have here.

Line 361 would then looks like:

```php
            // Provide info on user & reportback ids
             if (isset($options['includeUserIds']) && $options['includeUserIds']) {
                 $topThree[$key]['user_id'] = $user->id;
                 $topThree[$key]['reportback_id'] = $user->reportback->id;
             }
```